### PR TITLE
chore: Add support policy

### PR DESCRIPTION
**Pull request by farleyb-amazon**
*Description of changes:*
Add a dedicated page describing our support policy, and link to it from the README.

https://github.com/farleyb-amazon/aws-encryption-sdk-python/blob/support-policy/SUPPORT_POLICY.rst

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by antonf-amzn — APPROVED** [review]
lgtm

File: SUPPORT_POLICY.rst [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+      - Next status
+      - Next status date
+    * - 1.x
+      - Maintenance
+      - End of Support
+      - 2022-06-30
+    * - 2.x
+      - General Availability 
+      - Maintenance
+      - 2021-06-30
```
Comment (seebees) [inline SUPPORT_POLICY.rst:31]: I don't understand this date

Comment (lavaleri) [reply SUPPORT_POLICY.rst:31]: I'm reading this as 2.x is transitioning from General Availability -> Maintenance *today*. Should this be `2022-06-30`?

**Review by lavaleri — APPROVED** [review]

File: SUPPORT_POLICY.rst (hunk 1 above)
Comment (seebees) [reply SUPPORT_POLICY.rst]: Online conversation, this makes sense as we move to Python 2.x depreciation.

File: SUPPORT_POLICY.rst [hunk 2]
```diff
@@ -0,0 +1,37 @@
+Overview
+========
+This page describes the support policy for the AWS Encryption SDK. We regularly provide the AWS Encryption SDK with updates that may contain support for new or updated APIs, new features, enhancements, bug fixes, security patches, or documentation updates. Updates may also address changes with dependencies, language runtimes, and operating systems.
+
+We recommend users to stay up-to-date with Encryption SDK releases to keep up with the latest features, security updates, and underlying dependencies. Continued use of an unsupported SDK version is not recommended and is done at the user’s discretion
```
Comment (texastony) [inline SUPPORT_POLICY.rst:5]: Alas, we missed a `.`.

Comment (texastony) [inline SUPPORT_POLICY.rst:5]: Alas, we missed a period.

```suggestion
We recommend users to stay up-to-date with Encryption SDK releases to keep up with the latest features, security updates, and underlying dependencies. Continued use of an unsupported SDK version is not recommended and is done at the user’s discretion.
```

**Pull request merged**